src/types/dsnpTypes.ts

```diff
@@ -0,0 +1,103 @@
+import { EthereumAddress } from "@liberty30/lib-privacy-js";
+
+// Broadcast: a public post
+// fromAddress	ID of the sender	bytes20
+// messageID	keccak-256 hash of content stored at URI	bytes32
+// uri	content URI	string
+export interface DSNPBroadcast {
+  fromAddress: EthereumAddress;
+  messageID: Uint8Array;
+  uri: string;
+}
+
+// Reply: a public reply post
+// fromAddress	ID of the sender	bytes20
+// inReplyTo	ID of the message the reply references	bytes32
+// messageID	keccak-256 hash of content stored at uri	bytes32
+// uri	content uri	string
+export interface DSNPReply {
```

Comment:
Should these all extend a type of `DSNPMessage` (or other name)? Or perhaps build up a union type of all of them?

Reply:
Josh and I were talking about this and we thought that a "reply" was a comment on a post and "message" was a direct message.

Reply:
I namespaced it.   I realized the specs aren't consistent in naming as well.  There is now a union type for DSNPMessage that has all of them.

Reply:
@wilwade ^^